This patch fixes the shared library support in the target debugger and gdbserver.

Signed-off-by: Chris Smith <chris.smith@st.com>

Index: gdb-6.8/gdb/sh-linux-tdep.c
===================================================================
--- gdb-6.8.orig/gdb/sh-linux-tdep.c	2008-09-19 12:15:08.000000000 +0100
+++ gdb-6.8/gdb/sh-linux-tdep.c	2008-09-22 15:05:19.000000000 +0100
@@ -33,6 +33,8 @@
 #include "osabi.h"
 #include "symtab.h"
 #include "linux-tdep.h"
+#include "solib.h"
+#include "solist.h"
 
 #include "sh-tdep.h"
 #include "glibc-tdep.h"
@@ -339,6 +341,18 @@
   unwind_data->pcshift = 0;
   unwind_data->spshift = 9*4;
   frame_unwind_prepend_unwinder (gdbarch, &linux_context_frame_unwinder);
+
+  /* GNU/Linux uses SVR4-style shared libraries.  */
+  set_gdbarch_skip_trampoline_code (gdbarch, find_solib_trampoline_target);
+  set_solib_svr4_fetch_link_map_offsets
+                              (gdbarch, svr4_ilp32_fetch_link_map_offsets);
+
+  /* GNU/Linux uses the dynamic linker included in the GNU C Library.  */
+  set_gdbarch_skip_solib_resolver (gdbarch, glibc_skip_solib_resolver);
+
+  /* Enable TLS support.  */
+  set_gdbarch_fetch_tls_load_module_address (gdbarch,
+                                             svr4_fetch_objfile_link_map);
 }
 
 void
Index: gdb-6.8/gdb/configure.tgt
===================================================================
--- gdb-6.8.orig/gdb/configure.tgt	2008-09-19 12:15:08.000000000 +0100
+++ gdb-6.8/gdb/configure.tgt	2008-09-22 15:08:06.000000000 +0100
@@ -370,7 +370,8 @@
 sh*-*-linux*)
 	# Target: GNU/Linux Super-H
 	gdb_target_obs="sh-tdep.o sh64-tdep.o sh-linux-tdep.o monitor.o \
-			dsrec.o solib.o solib-svr4.o symfile-mem.o linux-tdep.o"
+			dsrec.o solib.o solib-svr4.o symfile-mem.o
+			linux-tdep.o glibc-tdep.o"
 	gdb_sim=
 	build_gdbserver=yes
 	;;
